ouroboros-consensus-test/bench-mempool/Main.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedLists    #-}
+
+
+module Main (main) where
+
+import           Control.Monad (void)
+import           Data.Set ()
+import           Test.Tasty.HUnit (testCase, (@?=))
+
+import qualified Control.Tracer as Tracer
+import           Test.Tasty.Bench (bench, bgroup, defaultMain, nfIO)
+
+import           Bench.Consensus.Mempool
+import           Bench.Consensus.Mempool.TestBlock (TestBlock)
+import qualified Bench.Consensus.Mempool.TestBlock as TestBlock
+import           Bench.Consensus.MempoolWithMockedLedgerItf
+import qualified Ouroboros.Consensus.Mempool.Capacity as Mempool
+
+main :: IO ()
+main = defaultMain [ benchmarkJustAddingTransactions ]
+  where
+    benchmarkJustAddingTransactions =
+        bgroup "Just adding transactions"
+          $  fmap benchAddNTxs [1_000, 10_000, 100_000, 1_000_000]
+          <>
+             [ testCase "Transactions are added" $ do
+                 let
+                   n   = 1000
+                   txs = mkNTryAddTxs 1000
+                 mempool    <- addNTxs txs
+                 mempoolTxs <- getTxs mempool
+                 mempoolTxs @?= txsAddedInCmds txs
+                 length mempoolTxs @?= n
+             ]
+      where
+        benchAddNTxs n = bench (show n) $ nfIO $ addNTxs' $ mkNTryAddTxs n
+        -- FIXME: check that we need forcing. Forcing the list does not seem to make a difference
+
+{-------------------------------------------------------------------------------
+  Adding TestBlock transactions to a mempool
+-------------------------------------------------------------------------------}
+
+addNTxs' :: [MempoolCmd TestBlock] -> IO ()
+addNTxs' = void . addNTxs
```

Comment:
I can't check now because the module containing `MemppolWithMockedLedgerItf` is not available on the branch, but could the `void` here be preventing some evaluation inside the mempool that we would want to force?

Reply:
I removed this function in favour of the new approach using `env`.